apis/projectcontour/v1/helpers.go

```diff
@@ -13,6 +13,8 @@
 
```

Comment:
One thing I'm not clear on is how subconditions get removed (e.g. the `HTTPProxy` has an error, then the spec is updated to fix the issue, so now the subcondition reporting that error should go away). Does each pass through the HTTPProxy processor clear out the `Valid` condition and start fresh?

Reply:
Yes, that's the idea. I think I may have missed it here, but the Valid condition also includes an observedGeneration field, that indicates which version of the object the status was generated from (standard Condition pattern) - I think it's far easier to just throw away the old condition and make a new one than try and parse the old one.